### PR TITLE
Implemented locale compatible `ScientificInput`

### PR DESCRIPTION
**Pull request by pyZerrenner**
Solves #310 and #602 
- added `toDouble` and `toString` methods to `ScientificInput` for handling locale-sensitive conversion between numbers and strings through the validator's `locale()`
- have the `valueFromText` and `textFromValue` use the above methods for conversion
- move text processing in `valueFromText` outside the `try` statement

### CODE REVIEW

**Comment by BenediktBurger** [comment]
Thanks for your pull request.
As a new contributor, you can add yourself (in this branch) to the `authors.txt`, such that, on merging, you will be named among the contributors!

@msmttchr  or @CasperSchippers could you take a look at that small change regarding parameters?

**Comment by pyZerrenner** [comment]
I added myself (Till Zürner, nice to meet you) to the author list.

**Comment by BenediktBurger** [comment]
Thanks.

I'm not that familiar with the data acquisition. Therefore, I ask you to be patient, until one of the aforementioned maintainers finds spare time for a review.

File: pymeasure/display/inputs.py
Comment (CasperSchippers) [inline pymeasure/display/inputs.py:251]: Just wondering: is there a specific reason to use the validator's locale, rather than the scientific input's locale (i.e., `self.locale()`)?

(same question of course for the toDouble)

**Review by CasperSchippers — COMMENTED** [review]
Thanks for this contribution, it is very much appreciated.

Generally looks good to me, just had a few minor questions.

Additionally, it might be good to add some tests that catch this behaviour.

File: pymeasure/display/inputs.py
Comment (pyZerrenner) [reply pymeasure/display/inputs.py:246]: I mainly added this, so that `toDouble` has the same behavior as the conversion with `float` when it fails. `valueFromText` still looks for a `ValueError` for when converting the input.

File: pymeasure/display/inputs.py
Comment (pyZerrenner) [reply pymeasure/display/inputs.py:260]: Sure, go ahead. Is the removal of units not necessary anymore?

File: pymeasure/display/inputs.py
Comment (pyZerrenner) [reply pymeasure/display/inputs.py:251]: I actually was not aware that `ScientificInput` itself (and all `QWidgets`?) have a locale. If it is guaranteed, that the validator has the same locale, then the other one can also be used. If there is some chance that they are different, using the locale of the validator, that sanitizes the input text, would be more fitting.

**Comment by pyZerrenner** [comment]
Thanks for writing the test. I am not familiar with that, so I would not know where to start.

File: pymeasure/display/inputs.py
Comment (CasperSchippers) [reply pymeasure/display/inputs.py:260]: Oh good one it is still needed. I thought I copied and adjusted your code for the suggestion, but it seems I failed to do that.

File: pymeasure/display/inputs.py
Comment (CasperSchippers) [reply pymeasure/display/inputs.py:251]: I'm not sure if there is a chance that the validator has a different locale... It makes sense indeed to ensure that we use the same as the validator, so let's just leave it as it is.

File: pymeasure/display/inputs.py
Comment (CasperSchippers) [reply pymeasure/display/inputs.py:260]: Wait, I did actually look at that, but got confused just now: it seems that the git suggestion thingy is playing up a bit and is comparing this suggestion to the old (pre-PR) code, not to your code. So the `text` has been sanatized by removing the units in your line 256 (which is indeed nicer than passing it directly to the conversion function.

Comment (CasperSchippers) [reply pymeasure/display/inputs.py]: Ok, the suggestion was not working at all.. anyways, fixed it.

**Review by CasperSchippers — APPROVED** [review]

**Comment by BenediktBurger** [comment]
Hey @CasperSchippers

As you approve it, you might merge it, I think.

I've seen, that you linked another issue. Good to see that older issue fixed as well. Thanks @pyZerrenner.

**Comment by CasperSchippers** [comment]
Yes, was waiting for the CI to finish.

Indeed very nice that it closes two pretty old issues 🎉

File: pymeasure/display/inputs.py
Comment (pyZerrenner) [reply pymeasure/display/inputs.py]: You are right, I forgot that I pulled the unit processing out of the `try`-block.

**Comment by pyZerrenner** [comment]
Thanks for your support and I am glad that I could contribute!

**Pull request merged**
